azsettings/cloud_settings.go

```diff
@@ -55,32 +53,48 @@ var predefinedClouds = []*AzureCloudSettings{
 	},
 }
 
+func (*AzureSettings) GetCloud(cloudName string) (*AzureCloudSettings, error) {
+	clouds := getClouds()
+
+	for _, cloud := range clouds {
+		if cloud.Name == cloudName {
+			return cloud, nil
+		}
+	}
+
+	return nil, fmt.Errorf("the Azure cloud '%s' not supported", cloudName)
+}
+
 func (*AzureSettings) Clouds() []AzureCloudInfo {
 	clouds := getClouds()
+	return mapCloudInfo(clouds)
+}
+
+func (*AzureSettings) ConfiguredClouds() []AzureCloudInfo {
```

Comment:
I'd prefer to call this `CustomClouds` to make it distinct this is only returning clouds that have been specified on the instance.